jboss/container/wildfly/launch/datasources/added/launch/datasource-common.sh

```diff
@@ -412,7 +414,7 @@ function generate_external_datasource_xml() {
     fi
   fi
 
-  if [ -n "$min_pool_size" ] || [ -n "$max_pool_size" ]; then
+  if [ -n "$min_pool_size" ] || [ -n "$max_pool_size" ] || [ -n "$is_same_rm_override" ] || [ -n "$no_tx_separate_pools" ]; then
     if [ -n "$NON_XA_DATASOURCE" ] && [ "$NON_XA_DATASOURCE" = "true" ]; then
        ds="$ds
```

Comment:
We now have 2 ways to configure the server: sed (legacy) and CLI operations. The wildfly-cekit-modules repository is targeting future releases (CD18, CD19) in which sed is no more used for new features.
You can see how we are using CLI to configure the server in generate_external_datasource_cli method. It would be the way to implement your RFE.